byway/src/main/webapp/routepage.js

```diff
@@ -11,171 +11,206 @@
 function loadData(){
   getRecsOnload();
   getStopsOnload();
 }
-
+ 
 /** Initializes map on the page */
 function initMap() {
-  let directionsService = new google.maps.DirectionsService();
-  let directionsRenderer = new google.maps.DirectionsRenderer();
-  let start = new google.maps.LatLng(37.7699298, -122.4469157);
-  let end = new google.maps.LatLng(37.7683909618184, -122.51089453697205);
+  directionsService = new google.maps.DirectionsService();
+  directionsRenderer = new google.maps.DirectionsRenderer();
+  start = "111 8th Ave, New York, NY";
+  end = "Yonkers, NY";
+ 
   let mapOptions = {
     zoom: 14,
-    center: start
+    center: new google.maps.LatLng(40.730610, -73.935242) // coordinates of NYC
   }
-  let map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  const map = new google.maps.Map(document.getElementById('map'), mapOptions);
   directionsRenderer.setMap(map);
-  document.getElementById("route").addEventListener("click", function() {
-    calcRoute(directionsService, directionsRenderer, start, end);
-  });
 }
-
+ 
 /** 
- * Displays route containing waypoints overtop the map.
- * TODO: Add in code from routeDir to change route in real-time
- * based on what's in the schedule panel
- * @param {DirectionsService} directionsService object that communicates with the GMaps API service
- * @param {DirectionsRenderer} directionsRenderer object that renders display results on the map
- * @param {String} start starting point of route
- * @param {String} end ending point of route. TODO: In the final implementation, just have 
- * start as start and end are the same
+ * Displays route containing waypoints of places the user wants to visit, 
+ * as shown in the schedule panel
  */
-function calcRoute(directionsService, directionsRenderer, start, end) {
+function calcRoute() {
   let request = {
     origin:  start,
     destination: end,
-    travelMode: 'DRIVING'
+    travelMode: 'DRIVING',
+    waypoints: Array.from(waypoints),
+    optimizeWaypoints: true
   };
   directionsService.route(request, function(response, status) {
     if (status == 'OK') {
       directionsRenderer.setDirections(response);
+      computeTotalDistance(response);
     } else {
       window.alert("Could not calculate route due to: " + status);
     }
   });
 }
 
+/**
+ * Calculates and sums up the distance and time duration between all destinations (legs)
+ * @param {response} result response from the directions service object
+ */
+function computeTotalDistance(result) {
+  var totalDist = 0;
+  var totalTime = 0;
+  // full route
+  var route = result.routes[0];
+  for (i = 0; i < route.legs.length; i++) {
+    totalDist += route.legs[i].distance.value;
+    totalTime += route.legs[i].duration.value;
+  }
+
+  totalDist = (totalDist / 1000).toFixed(2);
+  hours = Math.floor(totalTime / 3600);
+  minutes = Math.round((totalTime - hours*3600) / 60);
+ 
+  document.getElementById("distance").innerText = totalDist + "km";
```

Comment:
How are the units of distance determined? Does it default to meters?

Reply:
It says "km" in the text which I thought was sufficient, but I will comment for clarity since it does start as meters.

Reply:
This was more of a question about the API rather than a comment on your code. Just wondering if this is specified in your request or if it always returns meters. May be useful to toggle between kilometers and miles in the future if possible, but don't worry about it now.

Reply:
I added a comment saying the API defaults to meters.